src/cargo/ops/cargo_add/mod.rs

```diff
@@ -329,6 +338,26 @@ pub struct DepOp {
     pub tag: Option<String>,
 }
 
+fn spec_fix_suggestion(arg: &DepOp) -> Option<&'static str> {
+    let spec = arg.crate_spec.as_deref()?;
+
+    if arg.git.is_none()
+        && (spec.starts_with("git@") || spec.starts_with("ssh:") || spec.ends_with(".git"))
+    {
+        Some("git URLs must be specified with --git")
+    } else if arg.registry.is_none()
+        && (spec.starts_with("registry+") || spec.starts_with("sparse+"))
```

Comment:
`--registry` values do not start with those but are names to look up urls in config

Reply:
Updated the message